src/Tasks/Microsoft.NET.Build.Tasks/CheckForDuplicateItems.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public class CheckForDuplicateItems : TaskBase
+    {
+        [Required]
+        public ITaskItem [] Items { get; set; }
+
+        [Required]
+        public string ItemName { get; set; }
+
+        public bool DefaultItemsEnabled { get; set; }
+
+        public bool DefaultItemsOfThisTypeEnabled { get; set; }
+
+        public bool DefaultItemsWereIncluded { get; set; }
+
+        public string PropertyNameToDisableDefaultItems { get; set; }
+
+        public string PropertyValueToDisableDefaultItems { get; set; } = "false";
+
+        public string MoreInformationLink { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            var duplicateItems = Items.GroupBy(i => i.ItemSpec).Where(g => g.Count() > 1).ToList();
```

Comment:
Also is g.Count() cheap in practice?

I would be interested in seeing some measurment where this task shows up on the perf summary since it's doing work. Maybe it doesn't register in comparison to surrounding logic, but my instinct from lower level work is to cringe at all of this LINQ.